### PR TITLE
engine: docker compose build logs show up in manifest specific log

### PR DESCRIPTION
**Pull request by jazzdan**
Take 2!

### CODE REVIEW

File: internal/dockercompose/fake_client.go [hunk 1]
```diff
@@ -32,7 +36,9 @@ func (c *FakeDCClient) Down(ctx context.Context, pathToConfig string, stdout, st
 }
 
 func (c *FakeDCClient) StreamLogs(ctx context.Context, pathToConfig, serviceName string) (io.ReadCloser, error) {
-	return ioutil.NopCloser(bytes.NewReader([]byte(c.logOutput))), nil
+	output := c.RunLogOutput[model.ManifestName(serviceName)]
+	fmt.Printf("Got output: %s for manifest %s\n", output, serviceName)
```
Comment (nicks) [inline internal/dockercompose/fake_client.go]: ?

File: internal/engine/upper_test.go [hunk 2]
```diff
@@ -90,6 +92,10 @@ func (b *fakeBuildAndDeployer) nextBuildResult(ref reference.Named) store.BuildR
 }
 
 func (b *fakeBuildAndDeployer) BuildAndDeploy(ctx context.Context, manifest model.Manifest, state store.BuildState) (store.BuildResult, error) {
+	if manifest.IsDC() {
```
Comment (nicks) [inline internal/engine/upper_test.go]: why would this be specific to docker-compose?

File: internal/engine/upper_test.go [hunk 3]
```diff
@@ -2233,6 +2255,14 @@ func (f *testFixture) setupDCFixture() (redis, server model.Manifest) {
 	return manifests[0], manifests[1]
 }
 
+func (f *testFixture) setDCBuildLogOutput(mn model.ManifestName, output string) {
+	f.b.buildOutput[mn] = output
```
Comment (nicks) [inline internal/engine/upper_test.go]: again, this doesn't look docker-compose specific

Comment (jazzdan) [reply internal/engine/upper_test.go]: That's fair, I'll remove that restriction.

Comment (jazzdan) [reply internal/engine/upper_test.go]: (It's now removed)

File: internal/engine/upper_test.go [hunk 4]
```diff
@@ -74,6 +74,8 @@ type fakeBuildAndDeployer struct {
 
 	// Set this to simulate the build failing. Do not set this directly, use fixture.SetNextBuildFailure
 	nextBuildFailure error
+
+	buildOutput map[model.ManifestName]string
```
Comment (maiamcc) [inline internal/engine/upper_test.go]: consider `buildLogOutput` to match naming in fakeDcCli

File: internal/engine/upper_test.go [hunk 5]
```diff
@@ -1729,37 +1737,52 @@ func TestDockerComposeEventSetsStatus(t *testing.T) {
...
 	m, _ := f.setupDCFixture()
-	expected := "spoonerisms_1  | 2018-12-20T16:11:04.070480042Z yarn install v1.10."
-	f.dcc.SetLogOutput(expected + "\n")
+	expected := "yarn install"
+	f.setBuildLogOutput(m.ManifestName(), expected)
 
-	f.Start([]model.Manifest{m}, true)
-	f.waitForCompletedBuildCount(1)
+	f.loadAndStart()
+	f.waitForCompletedBuildCount(2)
```
Comment (maiamcc) [inline internal/engine/upper_test.go]: wait why are there two builds now?

File: internal/hud/renderer_test.go [hunk 6]
```diff
@@ -355,6 +355,11 @@ func TestRenderLogModal(t *testing.T) {
 					"Hi hello I'm a docker compose log",
 					time.Now().Add(time.Second*-12),
 				),
+				BuildHistory: []model.BuildStatus{
+					model.BuildStatus{
+						Log: []byte("Hi hello I'm a docker compose build log"),
```
Comment (maiamcc) [inline internal/hud/renderer_test.go]: v good dummy text 👍

File: internal/engine/upper_test.go (hunk 5 above)
Comment (jazzdan) [reply internal/engine/upper_test.go]: The docker compose fixture now has two services ("redis" and "server), before it only had one.

Comment (maiamcc) [reply internal/engine/upper_test.go]: oh. just call `f.Start([]model.Manifest{m}, true)`, we've already loaded above. there are two for the cases where they're needed/useful, but should probs just use one unless it's a specific test for >1 manifest.

**Pull request merged**